rules/static.go

```diff
@@ -826,6 +837,70 @@ func (r *DefaultRuleRenderer) StaticManglePreroutingChain(ipVersion uint8) *Chai
 	}
 }
 
+func (r *DefaultRuleRenderer) StaticManglePostroutingChain(ipVersion uint8) *Chain {
+	rules := []Rule{}
+
+	// Accept immediately if IptablesMarkAccept is set.  Our filter-FORWARD chain sets this for
+	// any packets that reach the end of that chain.  The principle is that we don't want to
+	// apply normal host endpoint policy to forwarded traffic.
+	rules = append(rules, r.acceptAlreadyAccepted()...)
```

Comment:
After a bit of thought, I think the "allow" action for the mangle table should be "RETURN" instead of "ACCEPT", at least by default.  The mangle table is not normally used for policy so its default action is unlikely to be deny (we didn't even have rules in this chain before your work so it can't be deny-by-default).

If the mangle table is in use, it's to tweak packets just before they leave the box so we should just fall through to any rules that the user has.

Reply:
I think this one is trickier than it looks!  We do already have IptablesMangleAllowAction, but

- it defaults to ACCEPT
- it was presumably intended for the pre-existing mangle usage, i.e. pre-DNAT
- to follow through your point above, we maybe need to plumb an allowAction parameter through the whole of the host dispatch and endpoint chains, and review that they work with RETURN.

I'll review the last point in more detail...

Reply:
Actually the plumbing detail for the last point looks fine.  So it's just a matter of whether the existing IptablesMangleAllowAction would go, or if we need an independent parameter.

Reply:
As agreed, I've changed the code to Return; hardcoded as such until a need arises to make it configurable.